### ISSUE TITLE
A way to reference quoted alias without `:raw`

### ISSUE THREAD

**Issue by nbardiuk** [jkk/honeysql#497]
I would like to reference a quoted alias without using `:raw`

for example in ORDER BY clause

```sql
SELECT column_name AS "some-alias" FROM b ORDER BY "some-alias" ASC
```

It could be achieved right now using `:raw` but it's not safe when alias comes from 3d party.

```clojure
(honey.sql/format {:select [[:column-name "some-alias"]]
                   :from :b
                   :order-by [[[:raw "\"some-alias\""]]]})
```

`:quoted` doesn't help because it doesn't convert kebab column names to underscores

```clojure
(honey.sql/format {:select [[:column-name "some-alias"]]
                   :from :b
                   :order-by [:some-alias]}
                  {:quoted true})
```
```sql
SELECT "column-name" AS "some-alias" FROM "b" ORDER BY "some-alias" ASC
```

`:quoted-snake` corrupts unerscores in aliases
```clojure
(honey.sql/format {:select [[:column-name "some-alias"]]
                   :from :b
                   :order-by [:some-alias]}
                  {:quoted-snake true})
```
```sql
SELECT column_name AS "some_alias" FROM b ORDER BY some_alias ASC
```

**Comment by seancorfield** [comment]
I'm leaning toward adding a Special Syntax for `:alias` so you can say `:order-by [[[:alias "some-alias"]]]` and it will render it using the exact same rules as for the alias portion of a `:select` etc.

```clojure
user=>  (honey.sql/format {:select [[:column-name "some-alias"]]
                     :from :b
                     :order-by [[[:alias "some-alias"]]]})
["SELECT column_name AS \"some-alias\" FROM b ORDER BY \"some-alias\" ASC"]
user=>  (honey.sql/format {:select [[:column-name "some-alias"]]
                     :from :b
                     :order-by [[[:alias "some-alias"]]]}
                    {:quoted true})
["SELECT \"column-name\" AS \"some-alias\" FROM \"b\" ORDER BY \"some-alias\" ASC"]
```

This change has been made on **develop** and is available for testing via git deps on the 2.4.9999-SNAPSHOT on Clojars.

**Comment by nbardiuk** [comment]
I've tested with integration tests on my project and it works as expected. Thank you 👍🏻 

Just to mention for others who will find this issue - the alias cannot be any string and needs to be sanitized:
* honeysql throws if the alias contains semicolon `;`
* PostgreSQL throws if the alias contains a character with code zero `"\0"`
other database can have more limitations
